ModulePresentationsForCAP/gap/ModulePresentationsForCAP.gi

```diff
@@ -2146,3 +2172,296 @@ end );
     end, 1000 );
 
 end );
+
+##
+## Random methods
+##
+
+##
+## Interpretation of n:
+## The number of rows and colms of the matrix of the object is less or equal to n.
+InstallGlobalFunction( ADD_RANDOM_OBJECT_LEFT,
+
+  function( category )
+
+    AddRandomObject( category,
+
+      function( C, n )
+        local nr_rows, nr_cols, mat, homalg_ring;
+
+        homalg_ring := category!.ring_for_representation_category;
+
+        if n = 0 then
+          
+          return ZeroObject( category );
+
+        fi;
+
+        nr_rows := Random( [ 1 .. n ] );
+
+        nr_cols := Random( [ 1 .. n ] );
+
+        mat := homalg_ring!.random_matrix_func( nr_rows, nr_cols );
+
+        return AsLeftPresentation( mat );
+
+      end );
+
+end );
+
+##
+InstallGlobalFunction( ADD_RANDOM_OBJECT_RIGHT,
+
+  function( category )
+
+    AddRandomObject( category,
+        function( category, n )
+        local homalg_ring, left_presentations, M;
+
+        homalg_ring := category!.ring_for_representation_category;
+
+        left_presentations := LeftPresentations( homalg_ring );
+
+        M := RandomObject( left_presentations, n );
+
+        return AsRightPresentation( Involution( UnderlyingMatrix( M ) ) );
+
+      end );
+
+end );
+
+##
+## Interpretation of n:
+## The number of relations of the range of the created random morphism is n.
+InstallGlobalFunction( ADD_RANDOM_MORPHISM_WITH_FIXED_SOURCE_LEFT,
+    function( category )
+      local homalg_ring;
+
+      homalg_ring := category!.ring_for_representation_category;
+
+      AddRandomMorphismWithFixedSource( category,
+        function( M, n )
+          local m, y, syz, x, u, U, e;
+          
+          if n = 0 then
```

Comment:
Why do you return the morphism in the zero object in this case?
The function should work fine also for `n=0` and return a well defined morphism into an object with 0-relations.

Reply:
I don't have a simple code that can search for a random non-zero morphism from presentation with relations to a free presentation. I am open to any suggestions or ideas.

Reply:
Your current code should already provide non-trivial examples such morphisms.

The general idea is: compose `SyzygiesOfColumns(m)` with some other random matrix.

Reply:
Yes, Thanks!. I think I excluded this situation without a careful thought. It can be handled without any problems using the general approach.

Reply:
I've updated this block.

Reply:
Perfect!